core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala

```diff
@@ -35,12 +35,33 @@ class TaskInfo(
      */
     val index: Int,
     val attemptNumber: Int,
+    /**
+     * The actual RDD partition ID in this task.
+     * The ID of the RDD partition is always same even task retries.
+     */
+    val partitionId: Int,
     val launchTime: Long,
     val executorId: String,
     val host: String,
     val taskLocality: TaskLocality.TaskLocality,
     val speculative: Boolean) {
 
+  def this(
+      taskId: Long,
+      /**
+       * The index of this task within its task set. Not necessarily the same as the ID of the RDD
+       * partition that the task is computing.
+       */
+      index: Int,
+      attemptNumber: Int,
+      launchTime: Long,
+      executorId: String,
+      host: String,
+      taskLocality: TaskLocality.TaskLocality,
+      speculative: Boolean) = {
+    this(taskId, index, attemptNumber, -1, launchTime, executorId, host, taskLocality, speculative)
+  }
```

Comment:
All use of `TaskInfo` in spark should pass the `partitionId` - leaving this only for backward compatibility for end users.

Reply:
Sure, I will change them.